db/model/bot.js

```diff
@@ -168,5 +168,19 @@ module.exports = function bot(seq, dataTypes) {
     });
   };
 
+  Bot.prototype.isWritableBy = function (who) {
```

Comment:
Is there a better way to reuse this fn? its the same ofr botAction, botData, room and roomType

Reply:
i had tried to do this but kept running into problems with it missing context

Reply:
Ok! Maybe we could create a spike to look into that?

@tausifmuzaffar